### DISCUSSION TITLE
Switching from WCF  having Multiple Self Hosted WCF services using NetNamed Pipes to CoreWCF

### DISCUSSION THREAD

**Discussion by pranith-BH** [CoreWCF/CoreWCF#1309] [Q&A]
Hi CoreWCF Community,

We are in process of Migrating of migrating our SelfHostedWCF which use NetNamedPipe Bindings to CoreWCF. We have more than one SelfHostedService listening at different Base Addresses. 

When we try to launch the second CoreWCF service , it says the Address already in use . What are the key differences in NetNamedPipes between WCF to CoreWCF.

![image](https://github.com/CoreWCF/CoreWCF/assets/98176975/4d385a70-4e71-4b29-a74a-7d71fc23903c)

**Comment by mconnew** [comment]
This is because aspnetcore is adding an http listening endpoint by default. Check your launchSettings.json and you should find they both projects are configured to listen on http/https on the same ports. Remove it from your launchSettings.json, or make each app use a different port number and that should fix the problem.

**Reply by mconnew** [reply]
It looks like even without a launchSettings.json it still listens on port 5000 by default (with .NET 6). There's a way to do this easily with .NET 8, but I'm trying to find a way to do this for .NET 6.
